packages/api/endpoints/collections.js

```diff
@@ -162,29 +169,38 @@ async function post(req, res) {
 async function put(req, res) {
   const { name, version } = req.params;
   const collection = req.body;
+  let dynamoRecord;
+  let oldCollection;
 
   if (name !== collection.name || version !== collection.version) {
     return res.boom.badRequest('Expected collection name and version to be'
       + ` '${name}' and '${version}', respectively, but found '${collection.name}'`
       + ` and '${collection.version}' in payload`);
   }
-
   const collectionsModel = new models.Collection();
-
-  if (!(await collectionsModel.exists(name, version))) {
+  try {
+    oldCollection = await collectionsModel.get({ name, version });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
+      throw error;
+    }
     return res.boom.notFound(
       `Collection '${name}' version '${version}' not found`
     );
   }
 
-  const dynamoRecord = await collectionsModel.create(collection);
+  collection.updatedAt = Date.now();
+  collection.createdAt = oldCollection.createdAt;
 
-  const dbRecord = dynamoRecordToDbRecord(dynamoRecord);
+  const postgresCollection = dynamoRecordToDbRecord(collection);
 
   const dbClient = await getKnexClient();
   await dbClient.transaction(async (trx) => {
-    await trx('collections').where({ name, version }).del();
-    await trx('collections').insert(dbRecord);
+    await trx(tableNames.collections)
```

Comment:
Functionally this accomplishes the same thing, courtesy of our explicit sets of 'undefined' on the keys of the merge object in the translate method.

Reply:
Yeah looks good, though this kind of DB interaction should be defined on the models per our recent discussions. I already had to make this change elsewhere so you can just merge it in:

https://github.com/nasa/cumulus/pull/2050/commits/73c7b157b7e5b90271ac523b719bcdae4e4742cf

Reply:
Pulled in/merged these changes as part of 90877b27b